src/Cli/dotnet/commands/dotnet-workload/install/WorkloadManifestUpdater.cs

```diff
@@ -76,59 +78,59 @@ private IEnumerable<ManifestId> GetInstalledManifestIds()
             string packagePath = null;
             string extractionPath = null;
+
+            var adManifestPath = GetAdvertisingManifestPath(_sdkFeatureBand, manifestId);
             try
             {
-                var adManifestPath = GetAdvertisingManifestPath(featureBand, manifestId);
-                packagePath = await _nugetPackageDownloader.DownloadPackageAsync(GetManifestPackageId(featureBand, manifestId), includePreview: includePreviews);
-                extractionPath = Path.Combine(_userHome, ".dotnet", "sdk-advertising-temp", $"{manifestId}-extracted");
-                Directory.CreateDirectory(extractionPath);
-                var resultingFiles = await _nugetPackageDownloader.ExtractPackageAsync(packagePath, extractionPath);
-
-                if (Directory.Exists(adManifestPath))
-                {
-                    Directory.Delete(adManifestPath, true);
-                }
-                Directory.CreateDirectory(Path.GetDirectoryName(adManifestPath));
-                FileAccessRetrier.RetryOnMoveAccessFailure(() => Directory.Move(Path.Combine(extractionPath, "data"), adManifestPath));
-
-                _reporter.WriteLine(string.Format(LocalizableStrings.AdManifestUpdated, manifestId));
+                packagePath = await _nugetPackageDownloader.DownloadPackageAsync(GetManifestPackageId(_sdkFeatureBand, manifestId), includePreview: includePreviews);
             }
-            catch (Exception e)
+            catch
             {
-                _reporter.WriteLine(string.Format(LocalizableStrings.FailedAdManifestUpdate, manifestId, e.Message));
+                _reporter.WriteLine(string.Format(LocalizableStrings.AdManifestPackageDoesNotExist, manifestId));
             }
-            finally
+            extractionPath = Path.Combine(_userHome, ".dotnet", "sdk-advertising-temp", $"{manifestId}-extracted");
+            Directory.CreateDirectory(extractionPath);
+            var resultingFiles = await _nugetPackageDownloader.ExtractPackageAsync(packagePath, extractionPath);
+
+            if (Directory.Exists(adManifestPath))
             {
-                if (!string.IsNullOrEmpty(extractionPath) && Directory.Exists(extractionPath))
-                {
-                    Directory.Delete(extractionPath, true);
-                }
+                Directory.Delete(adManifestPath, true);
+            }
+            Directory.CreateDirectory(Path.GetDirectoryName(adManifestPath));
+            FileAccessRetrier.RetryOnMoveAccessFailure(() => Directory.Move(Path.Combine(extractionPath, "data"), adManifestPath));
 
-                if (!string.IsNullOrEmpty(packagePath) && File.Exists(packagePath))
-                {
-                    File.Delete(packagePath);
-                }
+            _reporter.WriteLine(string.Format(LocalizableStrings.AdManifestUpdated, manifestId));
+
+            // Clean up temp dirs
+            if (!string.IsNullOrEmpty(extractionPath) && Directory.Exists(extractionPath))
```

Comment:
It looks like a bunch of this cleanup logic was moved out of a finally block, should we be cleaning up in the case of unexpected errors?  Does this fit into the transactions at all?

It's probably fine as is, I'm just wondering.